### PR TITLE
RPM packaging improvements

### PR DESCRIPTION
**Pull request by sachintu47**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Content Update

## Category

- [ ] zopen build framework
- [x] zopen package manager
- [ ] Documentation
- [x] CI/CD
- [ ] Tools

## Description

1. Replace cp -RP with pax -rw -p p in the install commands to preserve hardlinks during RPM build. cp -RP breaks hardlinks into independent copies, which inflates RPM size for packages like git that hardlink ~150+ commands to a single binary. With this fix, the git RPM is reduced from ~900MB to ~70MB.
2. Correct the install command in publish script

## Related Issues

- Related Issue #
- Closes #

### CODE REVIEW

**Comment by IgorTodorovskiIBM** [comment]
Here's some AI review comments. I'll defer to you to decide which are necessary

**Must fix**
(this is not introduced in this change)
**Source-root detection is not deterministic and can package the wrong subtree**

The logic selects the deepest directory named bin, lib, pyz, bash, or go:

FOUND_PATH=$(find . -type d \( -name bin -o -name lib -o -name pyz -o -name bash -o -name go \) ...)

This can select a nested directory rather than the actual package root. For example, if an archive contains:

usr/lpp/foo/bin/
usr/lpp/foo/lib/
usr/lpp/foo/lib/plugin/bin/

it may select usr/lpp/foo/lib/plugin/bin and package only that subtree, omitting the main binaries and libraries.

We should use a deterministic archive convention, an explicit root override, or derive a common package root rather than choosing the deepest matching directory.

**README.md is excluded completely**

It is skipped in the misc loop:

install_test.sh|metadata.json|setup.sh|test.status|README.md) continue ;;

and skipped again in the documentation loop:

case "$docfile" in
    README.md) continue ;;
esac

As a result, README.md is not included in the RPM.

**Other root-level documentation is copied twice**

Files such as LICENSE, COPYING, CHANGELOG, AUTHORS, and NEWS are copied first to %{_miscdir} and then again to %{_docdir}.

The misc loop should exclude all documentation patterns:

case "$item" in
  bin|altbin|lib|libexec|include|etc|share|man|doc|docs|scripts|data) continue ;;
  install_test.sh|metadata.json|setup.sh|test.status) continue ;;
  README*|LICENSE*|COPYING*|CHANGELOG*|AUTHORS*|INSTALL*|NEWS*) continue ;;
esac

Then the documentation loop can package them once under %{_docdir}.

**The FHS mapping changes the installation model and introduces package collisions**
(Igor: I think this is intended)

Previously, packages were isolated by name and version:

/opt/pkg/<name>/<version>

The new logic installs all packages into shared locations:

/opt/pkg/bin
/opt/pkg/lib
/opt/pkg/include
/opt/pkg/etc

This means packages can conflict on executable names, shared libraries, headers, config files, or common data files. It also removes the ability to have multiple versions installed side by side.

This may be the intended RPM model, but it needs an explicit design decision and a collision/versioning strategy before we make that change.

**The generated %files list has overlapping paths**

The file list can include both:

%{_libexecdir}
%{_libexecdir}/%{name}

Since the parent recursively owns the child, this is redundant and could lead to duplicate ownership issues. We should list only one of these paths.

**Additional concerns**
%config %{_sysconfdir} marks the entire shared /opt/pkg/etc directory as configuration. It would be safer to generate individual %config(noreplace) file entries.
The comments in analyze_pax_contents() about */bin/* not matching deep paths are misleading; shell globs can match those paths.
analyze_pax_contents() creates a temporary directory but does not use it.
We should add tests for hardlinks, symlinks, executable permissions, empty directories, root-level docs, package archives with nested bin directories, and two packages with colliding binaries or libraries.

**Comment by sachintu47** [comment]
augment review

File: bin/zopen-pax2rpm [hunk 1]
```diff
@@ -377,14 +479,24 @@ EOF
...
     [ "$has_scripts" = "true" ] && echo "  ✓ Scripts in scripts/"
-    
+    [ "$has_libexec" = "true" ] && echo "  ✓ Helper binaries in libexec/"
+    [ "$has_data" = "true" ] && echo "  ✓ Application data in data/"
+
+    echo ""
+    echo "=========================================="
+    echo "Generated spec file contents:"
+    echo "=========================================="
+    cat "$output_file"
```
[the remark this answers is not in the document]
Comment (sachintu47) [reply bin/zopen-pax2rpm]: Just added for testing, will be removed in the final commit

File: bin/zopen-pax2rpm [hunk 2]
```diff
@@ -248,38 +258,123 @@ if [ -d usr ]; then
...
+# Cleanup temp files
+rm -f "$_ALLFILES" "$_CFGFILES" "$_DOCFILES" "$_FILELIST.plain"
+
+# Warn if nothing was installed
+if [ "$(grep -c . "$_FILELIST" 2>/dev/null)" -le 1 ]; then
+    echo "WARNING: No files were installed into %{buildroot}. RPM payload will be empty." >&2
+fi
+
+echo "Generated RPM file list:"
+cat "$_FILELIST"
```
[the remark this answers is not in the document]
Comment (sachintu47) [reply bin/zopen-pax2rpm:391]: Just added for testing, will be removed in the final commit

**Comment by sachintu47** [comment]
augment review

**Comment by sachintu47** [comment]
augment review

**Comment by sachintu47** [comment]
augment review

File: cicd/build.groovy [hunk 3]
```diff
@@ -66,7 +66,7 @@ if [ "${GENERATE_PAX_RPM}" = "false" ]; then
 fi
 
 # Always run tests and update dependencies
-zopen build -v -b release -u $paxRpmOptions --no-set-active $extraOptions
+zopen build -v -b release -u $paxRpmOptions --no-set-active $extraOptions -sc
```
[the remark this answers is not in the document]
Comment (sachintu47) [reply cicd/build.groovy]: only enabled for test

**Comment by sachintu47** [comment]
augment review

**Comment by sachintu47** [comment]
augment review

**Comment by sachintu47** [comment]
augment review

**Comment by sachintu47** [comment]
augment review

**Comment by sachintu47** [comment]
augment review

File: bin/zopen-pax2rpm [hunk 4]
```diff
@@ -211,75 +215,140 @@ list_pax_contents() {
...
+        echo "Installing $dir/ -> $dest"
+        mkdir -p "$dest"
+        (cd "$dir" && pax -rw -p p . "$dest/") || { echo "ERROR: failed to install $dir/ -> $dest" >&2; exit 1; }
+    fi
+done
+
+# Install remaining files/directories that don't match known FHS dirs
+for item in *; do
+    case "$item" in
+        bin|altbin|lib|libexec|include|etc|share|man|doc|docs|scripts|data) continue ;;
```
[the remark this answers is not in the document]
Comment (sachintu47) [reply bin/zopen-pax2rpm:321]: should be okay. This will never happen

**Comment by sachintu47** [comment]
augment review
